src/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerWorkspaceFactory.cs

```diff
@@ -45,14 +45,20 @@ public LanguageServerWorkspaceFactory(
             .ToImmutableArray();
 
         // Create the workspace and set analyzer references for it
-        var workspace = new LanguageServerWorkspace(hostServicesProvider.HostServices);
+        var workspace = new LanguageServerWorkspace(hostServicesProvider.HostServices, WorkspaceKind.Host);
         workspace.SetCurrentSolution(s => s.WithAnalyzerReferences(CreateSolutionLevelAnalyzerReferencesForWorkspace(workspace)), WorkspaceChangeKind.SolutionChanged);
-        Workspace = workspace;
 
-        ProjectSystemProjectFactory = new ProjectSystemProjectFactory(
-            Workspace, fileChangeWatcher, static (_, _) => Task.CompletedTask, _ => { },
+        HostProjectFactory = new ProjectSystemProjectFactory(
+            workspace, fileChangeWatcher, static (_, _) => Task.CompletedTask, _ => { },
             CancellationToken.None); // TODO: do we need to introduce a shutdown cancellation token for this?
-        workspace.ProjectSystemProjectFactory = ProjectSystemProjectFactory;
+        workspace.ProjectSystemProjectFactory = HostProjectFactory;
+
+        var fbpWorkspace = new LanguageServerWorkspace(hostServicesProvider.HostServices, WorkspaceKind.MiscellaneousFiles);
+        fbpWorkspace.SetCurrentSolution(s => s.WithAnalyzerReferences(CreateSolutionLevelAnalyzerReferencesForWorkspace(fbpWorkspace)), WorkspaceChangeKind.SolutionChanged);
+
+        FileBasedProgramsProjectFactory = new ProjectSystemProjectFactory(
+            fbpWorkspace, fileChangeWatcher, static (_, _) => Task.CompletedTask, _ => { }, CancellationToken.None);
```

Comment:
what does it mean to be passing CancellationToken.None here.

Reply:
It means there is no support for tearing down this instance as a whole, unloading all the file watchers it created.